app/services/discourse_subscriptions/campaign.rb

```diff
@@ -0,0 +1,175 @@
+# frozen_string_literal: true
+
+module DiscourseSubscriptions
+  class Campaign
+    include DiscourseSubscriptions::Stripe
+
+    def initialize
+      set_api_key # instantiates Stripe API
+    end
+
+    def refresh_data
+      product_ids = Product.all.pluck(:external_id)
+
+      # if a product id is set for the campaign, we only want to return those results.
+      # if it's blank, return them all.
+      campaign_product = SiteSetting.discourse_subscriptions_campaign_product
+      product_ids = product_ids.select { |id| id == campaign_product } if campaign_product.present?
+
+      amount = 0
+      subscriptions = get_subscription_data
+      subscriptions = filter_to_subscriptions_products(subscriptions, product_ids)
+
+      # get number of subscribers
+      SiteSetting.discourse_subscriptions_campaign_subscribers = subscriptions&.length || 0
+
+      # calculate amount raised
+      subscriptions&.map do |sub|
+        sub_amount = calculate_monthly_amount(sub)
+        amount += sub_amount
+      end
+
+      SiteSetting.discourse_subscriptions_campaign_amount_raised = amount
+
+      if SiteSetting.discourse_subscriptions_campaign_show_contributors == true
+        contributor_ids = campaign_product.present? ? Customer.where(product_id: campaign_product).last(5).pluck(:user_id) : Customer.last(5).pluck(:user_id)
+        usernames = contributor_ids.map { |id| ::User.find(id).username }
+        SiteSetting.discourse_subscriptions_campaign_contributors = usernames.join(",") || ""
+      else
+        SiteSetting.discourse_subscriptions_campaign_contributors = ""
+      end
+    end
+
+    def create_campaign
+      begin
+        group = create_campaign_group
+        product = create_campaign_product
+        create_campaign_prices(product, group)
+
+        SiteSetting.discourse_subscriptions_campaign_enabled = true
+        SiteSetting.discourse_subscriptions_campaign_product = product[:id]
+      rescue ::Stripe::InvalidRequestError => e
+        e
+      end
+    end
+
+    protected
+
+    def create_campaign_group
+      # since this is public, we want to localize this as much as possible
+      group = ::Group.find_by(name: I18n.t('js.discourse_subscriptions.campaign.supporters'))
```

Comment:
This won't work if that translation key is changed. You should store the `group_id` of the created group instead, probably in a table or hidden site setting.